packages/@jsii/go-runtime/jsii-runtime-go/helpers.go

```diff
@@ -31,21 +53,13 @@ func String(v string) *string { return &v }
 
 // Strings obtains a pointer to a slice of pointers to all the provided strings.
 func Strings(v ...string) *[]*string {
-	slice := make([]*string, len(v))
-	for i := 0; i < len(v); i++ {
-		slice[i] = String(v[i])
-	}
-	return &slice
+	return Slice(v...)
 }
 
 // Time obtains a pointer to the provided time.Time.
 func Time(v time.Time) *time.Time { return &v }
```

Comment:
I wonder if these should be re-implemented to use `V`, so there's only 1 concrete implementation here... I'd be hopeful the compiler ends up being smart enough to inline `V`, so it ultimately makes no performance difference? (and if it did, would be be noticeable?)

Reply:
I thought that they were so simple that they could be happily left alone, and that it would be simpler to do so.

My understanding is that Go 1.18 was pretty weak on monomorphization and that generics add a runtime cost - https://planetscale.com/blog/generics-can-make-your-go-code-slower but, based on your comment I did a quick benchmark and given these two:

```go
package genericptr

import "time"

type types interface {
	bool | string | float64 | time.Time
}

func Ptr[T types](v T) *T { return &v }

func String(v string) *string { return &v }
```

And this benchmark...

```go
package genericptr

import "testing"

func BenchmarkPtr(b *testing.B) {
	for i := 0; i < b.N; i++ {
		ss := Ptr("test")
		if *ss != "test" {
			b.Error("something went wrong")
		}
	}
}

func BenchmarkString(b *testing.B) {
	for i := 0; i < b.N; i++ {
		ss := String("test")
		if *ss != "test" {
			b.Error("something went wrong")
		}
	}
}
```

The difference was always < 10ns on multiple runs.

```
go test -bench=.
goos: linux
goarch: amd64
pkg: genericptr
cpu: 11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz
BenchmarkPtr-16         1000000000               0.2092 ns/op
BenchmarkString-16      1000000000               0.2090 ns/op
PASS
```

Reply:
Based on that, I agree - single implementation.